cmd/jaeger/internal/extension/remotesampling/extension.go

```diff
@@ -0,0 +1,174 @@
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot find storage factory: %w", err)
+	}
+
+	ssStore, ok := f.(storage.SamplingStoreFactory)
+	if !ok {
+		return nil, nil, fmt.Errorf("storage factory of type %s does not support sampling store", samplingStorage)
+	}
+
+	lock, err := ssStore.CreateLock()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	store, err := ssStore.CreateSamplingStore(opts.AggregationBuckets)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ep := leaderelection.NewElectionParticipant(lock, defaultResourceName, leaderelection.ElectionParticipantOptions{
+		FollowerLeaseRefreshInterval: opts.FollowerLeaseRefreshInterval,
+		LeaderLeaseRefreshInterval:   opts.LeaderLeaseRefreshInterval,
+		Logger:                       logger,
+	})
+
+	return store, ep, nil
+}
+
+func (ext *RsExtension) Start(ctx context.Context, host component.Host) error {
+	if ext.Cfg.File.Path != "" {
+		opts := static.Options{
+			StrategiesFile: ext.Cfg.File.Path,
+		}
+
+		// TODO contextcheck linter complains about next line that context is not passed. It is not wrong.
+		//nolint
+		ss, err := static.NewStrategyStore(opts, ext.telemetry.Logger)
+		if err != nil {
+			return fmt.Errorf("failed to create the local file strategy store: %w", err)
+		}
+
+		ext.strategyStore = ss
+	}
+
+	if ext.Cfg.Adaptive.StrategyStore != "" {
+		opts := adaptive.Options{
+			InitialSamplingProbability:   ext.Cfg.Adaptive.InitialSamplingProbability,
+			MinSamplesPerSecond:          ext.Cfg.Adaptive.MinSamplesPerSecond,
+			LeaderLeaseRefreshInterval:   ext.Cfg.Adaptive.LeaderLeaseRefreshInterval,
+			FollowerLeaseRefreshInterval: ext.Cfg.Adaptive.FollowerLeaseRefreshInterval,
+			AggregationBuckets:           ext.Cfg.Adaptive.AggregationBuckets,
+		}
+
+		store, ep, err := GetSamplingStorageFactory(ext.Cfg.Adaptive.StrategyStore, host, opts, ext.telemetry.Logger)
+		if err != nil {
+			return err
+		}
+
+		ss := adaptive.NewStrategyStore(opts, ext.telemetry.Logger, ep, store)
+		ss.Start()
+		ext.strategyStore = ss
+	}
+
+	if err := ext.startHTTPServer(ctx, host); err != nil {
+		return fmt.Errorf("failed to start sampling http server: %w", err)
+	}
+
+	return nil
+}
+
+func (ext *RsExtension) startHTTPServer(ctx context.Context, host component.Host) error {
+	httpMux := http.NewServeMux()
+
+	handler := clientcfghttp.NewHTTPHandler(clientcfghttp.HTTPHandlerParams{
+		ConfigManager:  &clientcfghttp.ConfigManager{SamplingStrategyStore: ext.strategyStore},
+		MetricsFactory: metrics.NullFactory,
+		BasePath:       "/api",
+	})
+
+	handler.RegisterRoutesWithHTTP(httpMux)
+
+	var err error
+	if ext.httpServer, err = ext.Cfg.HTTP.ToServer(ctx, host, ext.telemetry, httpMux); err != nil {
+		return err
+	}
+
+	ext.telemetry.Logger.Info("Starting HTTP server", zap.String("endpoint", ext.Cfg.HTTP.ServerConfig.Endpoint))
+	var hln net.Listener
+	if hln, err = ext.Cfg.HTTP.ServerConfig.ToListener(ctx); err != nil {
```

Comment:
does OTEL collector ever allow multiple handlers to be attached to a single server? In Jaeger-v1 we often reused many ports for different services, e.g. for both submitting traces into backend and for retrieving sampling strategies.